modin/pandas/test/dataframe/test_default.py

```diff
@@ -1171,3 +1172,16 @@ def test_hasattr_sparse(is_sparse_data):
         else create_test_dfs(test_data["float_nan_data"])
     )
     eval_general(modin_df, pandas_df, lambda df: hasattr(df, "sparse"))
+
+
+def test_setattr_axes():
+    # Test that setting .index or .columns does not warn
+    df = pd.DataFrame([[1, 2], [3, 4]])
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
```

Comment:
```suggestion
        # In BaseOnPython, setting columns raises a warning because get_axis
        # defaults to pandas.
        if get_current_execution() != "BaseOnPython"
                warnings.simplefilter("error")
```

This should fix the CI failure in BaseOnPython: https://github.com/modin-project/modin/runs/7545798761?check_suite_focus=true